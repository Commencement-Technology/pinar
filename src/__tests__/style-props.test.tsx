import { shallow } from "enzyme";
import React from "react";
import { Text, View } from "react-native";
import Carousel from "../index";

describe("style props", (): void => {
  it("should allow style prop", (): void => {
    const wrapper = shallow(
      <Carousel style={{ backgroundColor: "blue" }}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should allow containerStyle prop (ScrollView)", (): void => {
    const wrapper = shallow(
      <Carousel containerStyle={{ backgroundColor: "blue" }}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should allow contentContainerStyle prop (ScrollView)", (): void => {
    const wrapper = shallow(
      <Carousel contentContainerStyle={{ backgroundColor: "blue" }}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should allow custom controlsContainerStyle", (): void => {
    const wrapper = shallow(
      <Carousel controlsContainerStyle={{ backgroundColor: "blue" }}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should allow custom controlsButtonStyle", (): void => {
    const wrapper = shallow(
      <Carousel controlsButtonStyle={{ backgroundColor: "blue" }} index={1}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
        <View>
          <Text>3</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should allow custom controlsTextStyle", (): void => {
    const wrapper = shallow(
      <Carousel controlsTextStyle={{ color: "blue" }} index={1}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
        <View>
          <Text>3</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should allow custom dotsContainerStyle", (): void => {
    const wrapper = shallow(
      <Carousel dotsContainerStyle={{ backgroundColor: "blue" }}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should allow custom dotStyle", (): void => {
    const wrapper = shallow(
      <Carousel dotStyle={{ backgroundColor: "blue" }}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should allow custom activeDotStyle", (): void => {
    const wrapper = shallow(
      <Carousel activeDotStyle={{ backgroundColor: "blue" }}>
        <View>
          <Text>1</Text>
        </View>
        <View>
          <Text>2</Text>
        </View>
      </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
