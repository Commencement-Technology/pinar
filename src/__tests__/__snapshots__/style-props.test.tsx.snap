// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`style props should allow containerStyle prop (ScrollView) 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "maxHeight": undefined,
        "maxWidth": undefined,
      },
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    >
      <View
        accessible={true}
        key="00"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="11"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View />
      <ForwardRef
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessible={true}
        onPress={[Function]}
        testID="PinarNextButton"
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          ›
        </Text>
      </ForwardRef>
    </View>
  </View>
</View>
`;

exports[`style props should allow contentContainerStyle prop (ScrollView) 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "maxHeight": undefined,
        "maxWidth": undefined,
      },
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      contentContainerStyle={
        Object {
          "backgroundColor": "blue",
        }
      }
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="00"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="11"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View />
      <ForwardRef
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessible={true}
        onPress={[Function]}
        testID="PinarNextButton"
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          ›
        </Text>
      </ForwardRef>
    </View>
  </View>
</View>
`;

exports[`style props should allow custom activeDotStyle 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "maxHeight": undefined,
        "maxWidth": undefined,
      },
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="00"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="11"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "blue",
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View />
      <ForwardRef
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessible={true}
        onPress={[Function]}
        testID="PinarNextButton"
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          ›
        </Text>
      </ForwardRef>
    </View>
  </View>
</View>
`;

exports[`style props should allow custom controlsButtonStyle 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "maxHeight": undefined,
        "maxWidth": undefined,
      },
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="00"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="11"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="22"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".2"
        >
          <Text>
            3
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="2/.2"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <ForwardRef
        accessibilityLabel="Previous"
        accessibilityRole="button"
        accessible={true}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "blue",
          }
        }
        testID="PinarPrevButton"
      >
        <Text
          accessibilityLabel="Previous"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          ‹
        </Text>
      </ForwardRef>
      <ForwardRef
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessible={true}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "blue",
          }
        }
        testID="PinarNextButton"
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          ›
        </Text>
      </ForwardRef>
    </View>
  </View>
</View>
`;

exports[`style props should allow custom controlsContainerStyle 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "maxHeight": undefined,
        "maxWidth": undefined,
      },
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="00"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="11"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    >
      <View />
      <ForwardRef
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessible={true}
        onPress={[Function]}
        testID="PinarNextButton"
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          ›
        </Text>
      </ForwardRef>
    </View>
  </View>
</View>
`;

exports[`style props should allow custom controlsTextStyle 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "maxHeight": undefined,
        "maxWidth": undefined,
      },
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="00"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="11"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="22"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".2"
        >
          <Text>
            3
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="2/.2"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <ForwardRef
        accessibilityLabel="Previous"
        accessibilityRole="button"
        accessible={true}
        onPress={[Function]}
        testID="PinarPrevButton"
      >
        <Text
          accessibilityLabel="Previous"
          accessible={true}
          style={
            Object {
              "color": "blue",
            }
          }
        >
          ‹
        </Text>
      </ForwardRef>
      <ForwardRef
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessible={true}
        onPress={[Function]}
        testID="PinarNextButton"
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "blue",
            }
          }
        >
          ›
        </Text>
      </ForwardRef>
    </View>
  </View>
</View>
`;

exports[`style props should allow custom dotStyle 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "maxHeight": undefined,
        "maxWidth": undefined,
      },
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="00"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="11"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "blue",
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View />
      <ForwardRef
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessible={true}
        onPress={[Function]}
        testID="PinarNextButton"
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          ›
        </Text>
      </ForwardRef>
    </View>
  </View>
</View>
`;

exports[`style props should allow custom dotsContainerStyle 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "maxHeight": undefined,
        "maxWidth": undefined,
      },
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="00"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="11"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View />
      <ForwardRef
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessible={true}
        onPress={[Function]}
        testID="PinarNextButton"
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          ›
        </Text>
      </ForwardRef>
    </View>
  </View>
</View>
`;

exports[`style props should allow style prop 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "maxHeight": undefined,
        "maxWidth": undefined,
      },
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }
>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      bounces={false}
      horizontal={true}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollEnabled={true}
      scrollEventThrottle={16}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View
        accessible={true}
        key="00"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".0"
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        key="11"
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          key=".1"
        >
          <Text>
            2
          </Text>
        </View>
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        key="0/.0"
        style={
          Object {
            "backgroundColor": "#333",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
      <View
        key="1/.1"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": 4,
            "height": 8,
            "marginBottom": 3,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 3,
            "width": 8,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View />
      <ForwardRef
        accessibilityLabel="Next"
        accessibilityRole="button"
        accessible={true}
        onPress={[Function]}
        testID="PinarNextButton"
      >
        <Text
          accessibilityLabel="Next"
          accessible={true}
          style={
            Object {
              "color": "#333",
              "fontSize": 50,
              "padding": 20,
            }
          }
        >
          ›
        </Text>
      </ForwardRef>
    </View>
  </View>
</View>
`;
